tests/cloudpickle_test.py

```diff
@@ -2352,6 +2397,77 @@ def test_lookup_module_and_qualname_stdlib_typevar():
     assert name == 'AnyStr'
 
 
+def test_register_pickle_by_value():
+    import _cloudpickle_testpkg
+    T = _cloudpickle_testpkg.T
+
+    try:
+        # Test that adding and removing a module to pickle by value returns
+        # us to the original pickle by reference
+        assert len(_PICKLE_BY_VALUE_MODULES) == 0
+        register_pickle_by_value(_cloudpickle_testpkg)
+        unregister_pickle_by_value(_cloudpickle_testpkg)
+        pickle_by_ref = _should_pickle_by_reference(T, name=T.__name__)
```

Comment:
This check currently raises an error since we did not implement handling of `TypeVar` instances in `_should_pickle_by_reference` (we bypassed its use in `_typevar_reduce`). To check that the registration procedure indeed work, you may need to do some pattern matching in the byte strings return by `cloudpickle.dumps` as discussed for `fuctions`.

Reply:
Would the solution be to add in an explicit typevar check in the `_should_pickle_by_reference`. Im not too sure why `_should_pickle_by_reference` is being dispatched for a TypeVar object (I thought it would hit the reduce when it delves into the objects inside the module)

Reply:
No, this would not be the right check: since we do not use use `_should_pickle_by_reference` in `_typevar_reduce`, this test currently test a function call that is never done in practice by `cloudpickle` itself. As explained, we need to look directly at the contents of the byte string in this case to make sure that `TypeVar` objects are pickled by value/by reference as expected.